mubench.pipeline/data/detector_execution.py

```diff
@@ -69,13 +65,11 @@ def __init__(self, run_mode: DetectorMode, detector: Detector, version: ProjectV
         self.findings_filter = findings_filter
 
     def execute(self, compile_base_path: str, timeout: Optional[int], logger: Logger):
-        detector_invocation = ["java"] + self.detector.java_options + ["-jar", _quote(self.detector.jar_path)]
-        command = detector_invocation + self._get_detector_arguments(self.version.get_compile(compile_base_path))
-        command = " ".join(command)
+        detector_args = self._get_detector_arguments(self.version.get_compile(compile_base_path))
 
         start = time.time()
         try:
-            Shell.exec(command, logger=logger, timeout=timeout)
+            self.detector.runner_interface.execute(self.version, detector_args, timeout, logger)
```

Comment:
This crashed with a null reference, when the interface has been set to None in line 28 of detector.py. Can we, instead of None, use a `NoInterface` implementation that raises a more meaningful exception on execute?